### PR TITLE
update(developer-tools): add new content

### LINKED ISSUE #262: Update Developer Tools Docs
After our session with @tetogomez on the Tech stack we will be working with lets update our favorite tools section

https://guide.eoscostarica.io/docs/developer-tools

- [x] Apollo 
- [x] Axios
- [x] PropTypes
- [x] Hapi

### PR DESCRIPTION
**Pull request by Edgar097**
### GH Issue

Resolves #262  

### Steps to test
1. Open Our Favorite Tools 
1. Check Content

#### CheckList
- [ ] Follow proper Markdown format
- [ ] The content is adequate
- [ ] The content is available in both english and spanish
- [ ] I Ran a spell check

### CODE REVIEW

**Comment by steph222** [comment]
@Edgar097  add this information 😄 

## Apollo

###  What is  Apollo?

Apollo is a suite of tools to create a GraphQL server, and to consume a GraphQL API.  Apollo is a team and community that builds on top of GraphQL, and provides different tools that help you build your projects.

### Why we use  Apollo?

1. **Documentation:** While Apollo continues to evolve, the team and community behind it keeps the documentation up to date, and they have plenty of insight about how to build applications. 
2. **Libraries:** Apollo offers plenty of libraries for implementing an effective GraphQL tech stack for JavaScript applications, and their libraries are open-sourced to be more manageable.
3.  **Features:** Apollo comes with built-in features to pull all the complexity out of applications and handle the intersection between client and server applications.

### Instalation of Apollo

[Apollo CLI Instalation](https://www.apollographql.com/docs/devtools/cli/)

[Apollo GitHub](https://github.com/apollographql)

* * *

## Axios

###  What is Axios?

Axios is a very popular JavaScript library you can use to perform HTTP requests, that works in both Browser and Node.js platforms. It supports all modern browsers, including support for IE8 and higher.

### Why we use Axios?

1. Supports older browsers.
2. Has a way to abort a request.
3. Has a way to set a response timeout.
4. has built-in CSRF protection.
5. Supports upload progress.
6. Performs automatic JSON data transformation.
7. Works in Node.js

### Instalation of Axios

[Axios Instalation](https://www.npmjs.com/package/axios)

[Axios GitHub](https://github.com/axios/axios)

File: docs/developer-tools.md [hunk 1]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
 
-https://github.com/facebook/react
+#### **PropTypes**
+
+Javascript is not a strongly typed language, this might create some problems for the developers. PropTypes is a library that helps when you need to resolve a problem related with this. Checking the types of the passed props object against a specificaction set beforehand. [PropTypes Documentation](https://reactjs.org/docs/typechecking-with-proptypes.html)
+
+### Why we use React JS?
+
+1. **Fast Learning Curve**. React is very a simple and lightweight library that only deals with the view layer.
+2. **Reusable Components**. React provides a component based structure. Components are your lego pieces. You start with tiny components like button, checkbox, dropdown etc. and the you create wrapper components composed of those smaller components.
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:80]: > checkbox, dropdown etc. and the you create

Not sure about that point

File: docs/developer-tools.md [hunk 2]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
 Modern command line HTTP client – user-friendly curl alternative with intuitive UI, JSON support, syntax highlighting, wget-like downloads, extensions, etc.
 
-https://github.com/jakubroztocil/httpie
+### Why we use httpie?
+
+1. **Simple**, expressive and intuitive HTTP command syntax.
+2. **Nicely formatted** and colorized terminal output.
+3. **Out-of-box** JSON support.
+4. **Forms** and **file uploads**.
+5. **Support**, for custom HTTP header and supports arbitrary request data..
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:153]: > request data..

Double end point

File: docs/developer-tools.md [hunk 3]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
+6. **Major OS supports** — Linux, macOS, and Windows.
+7. **Plugins** to support additional feature.
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:155]: We could use `:` or `,` but use the same one for the whole context

File: docs/developer-tools.md [hunk 4]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
 It enables developers to test calls to APIs. Postman users enter data. The data is sent to a special web server address. Typically, information is returned, which Postman presents to the user.
 
-https://github.com/postmanlabs
+### Why we use PostMan?
+
+1. **User-Friendly:** Having a user-friendly interface enables one to send a simple request by only entering it, filling in the necessary headers, selecting the HTTP method and pressing the ‘send’ button. 
+2. **Automation:**  Automated tests can be easily set up using the Postman API testing tool. You can automate your test cases in Postman in javascript with impressive features like write test suites, build request that consists of dynamic parameters, pass on data in between the request, etc. In this way tests and requests can be aggregated into a single automated test sequence.
+3. **Cost Effective:**  Apart from the many benefits which this tool provides us, this tool is also very cost effective. You can use the Postman in their free plan, which consists of limited features but most helpful.
+
+### PostMan Installation
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:203]: PostMan -> Postman

File: docs/developer-tools.md [hunk 5]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
+
+[PostMan Installation](https://www.softwaretestingmaterial.com/install-postman/)
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:205]: PostMan -> Postman

File: docs/developer-tools.md [hunk 6]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
+
+[PostMan GitHub](https://github.com/postmanlabs)
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:207]: PostMan -> Postman

File: docs/developer-tools.md [hunk 7]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
+
+1. **Simple**, expressive and intuitive HTTP command syntax.
+2. **Nicely formatted** and colorized terminal output.
+3. **Out-of-box** JSON support.
+4. **Forms** and **file uploads**.
+5. **Support**, for custom HTTP header and supports arbitrary request data..
+6. **Major OS supports** — Linux, macOS, and Windows.
+7. **Plugins** to support additional feature.
+
+### httpie Installation
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:157]: httpie -> HTTPie

File: docs/developer-tools.md [hunk 8]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
+
+### Hasura Installation
+
+[Hasura CLI Installation](https://hasura.io/docs/1.0/graphql/core/hasura-cli/install-hasura-cli.html).
+
+[Hsura GitHub](https://github.com/hasura)
 
 * * *
 
 ## httpie
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:141]: httpie -> HTTPie

File: docs/developer-tools.md [hunk 9]
```diff
@@ -6,64 +6,205 @@ sidebar_label: Our Favorite Tools
...
+
+1. Helps encapsulate the details of the **structure** of your tables behind a consistent interface. It also lets you structure your data in an intuitive way.
+2. Provides more **readability** since a named query can set the context much faster than a raw SQL query. You can use a view instead of littering your client code base with complex queries.
+3. Views have separate permissions and hence can be used to **restrict access** to the tables such that users are only allowed to see specific rows and columns.
+
+### Hasura Installation
+
+[Hasura CLI Installation](https://hasura.io/docs/1.0/graphql/core/hasura-cli/install-hasura-cli.html).
+
+[Hsura GitHub](https://github.com/hasura)
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:137]: Hsura GitHub - > Hasura GitHub

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

**Comment by Edgar097** [comment]
> @Edgar097 add this information 😄
> 
> ## Apollo
> ### What is Apollo?
> Apollo is a suite of tools to create a GraphQL server, and to consume a GraphQL API. Apollo is a team and community that builds on top of GraphQL, and provides different tools that help you build your projects.
> 
> ### Why we use Apollo?
> 1. **Documentation:** While Apollo continues to evolve, the team and community behind it keeps the documentation up to date, and they have plenty of insight about how to build applications.
> 2. **Libraries:** Apollo offers plenty of libraries for implementing an effective GraphQL tech stack for JavaScript applications, and their libraries are open-sourced to be more manageable.
> 3. **Features:** Apollo comes with built-in features to pull all the complexity out of applications and handle the intersection between client and server applications.
> 
> ### Instalation of Apollo
> [Apollo CLI Instalation](https://www.apollographql.com/docs/devtools/cli/)
> 
> [Apollo GitHub](https://github.com/apollographql)
> 
> ## Axios
> ### What is Axios?
> Axios is a very popular JavaScript library you can use to perform HTTP requests, that works in both Browser and Node.js platforms. It supports all modern browsers, including support for IE8 and higher.
> 
> ### Why we use Axios?
> 1. Supports older browsers.
> 2. Has a way to abort a request.
> 3. Has a way to set a response timeout.
> 4. has built-in CSRF protection.
> 5. Supports upload progress.
> 6. Performs automatic JSON data transformation.
> 7. Works in Node.js
> 
> ### Instalation of Axios
> [Axios Instalation](https://www.npmjs.com/package/axios)
> 
> [Axios GitHub](https://github.com/axios/axios)

Thanks!

**Review by xavier506 — DISMISSED** [review]
LGTM

File: docs/developer-tools.md [hunk 10]
```diff
@@ -4,66 +4,251 @@ title: Our Favorite Tools
 sidebar_label: Our Favorite Tools
 ---
 
+## Apollo
+
+### What is  Apollo?
+
+Apollo is a suite of tools to create a GraphQL server, and to consume a GraphQL API.  Apollo is a team and community that builds on top of GraphQL, and provides different tools that help you build your projects.
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:11]: > GraphQL server, and to consume

Not sure about this comma

File: docs/developer-tools.md [hunk 11]
```diff
@@ -4,66 +4,251 @@ title: Our Favorite Tools
...
 ## React JS
 
+### What is React JS?
+
 React is an open-source JavaScript library for building user interfaces. It is maintained by Facebook and a community of individual developers and companies. React can be used as a base in the development of single-page or mobile applications.
 
-https://github.com/facebook/react
+#### **PropTypes**
+
+Javascript is not a strongly typed language, this might create some problems for the developers. PropTypes is a library that helps when you need to resolve a problem related with this. Checking the types of the passed props object against a specificaction set beforehand. [PropTypes Documentation](https://reactjs.org/docs/typechecking-with-proptypes.html)
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:119]: specificaction -> specification

File: docs/developer-tools.md [hunk 12]
```diff
@@ -4,66 +4,251 @@ title: Our Favorite Tools
...
+
+### What is Postman?
 
 It enables developers to test calls to APIs. Postman users enter data. The data is sent to a special web server address. Typically, information is returned, which Postman presents to the user.
 
-https://github.com/postmanlabs
+### Why we use Postman?
+
+1. **User-Friendly:** Having a user-friendly interface enables one to send a simple request by only entering it, filling in the necessary headers, selecting the HTTP method and pressing the ‘send’ button. 
+2. **Automation:**  Automated tests can be easily set up using the Postman API testing tool. You can automate your test cases in Postman in javascript with impressive features like write test suites, build request that consists of dynamic parameters, pass on data in between the request, etc. In this way tests and requests can be aggregated into a single automated test sequence.
```
Comment (leisterfrancisco) [inline docs/developer-tools.md:244]: javascript  -> JavaScript

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

**Review by leisterfrancisco — APPROVED** [review]
Good work 👍

**Review by steph222 — APPROVED** [review]
Good Job 💯

**Pull request merged**